File: scala/leetcode/src/main/scala/leetcode/editor/cn/[27]移除元素.scala
package leetcode.editor.cn
object RemoveElement {
//leetcode submit region begin(Prohibit modification and deletion)
object Solution {
    def removeElement(nums: Array[Int], `val`: Int): Int = {
      var slow = 0
      var fast = 0
      while (fast < nums.length) {
        if (nums(fast) != `val`){
          nums(slow) = nums(fast)
          slow += 1
        }
        fast += 1
      }
      slow
    }
}
//leetcode submit region end(Prohibit modification and deletion)

}

File: cpp/leetcodecn/[27]移除元素.cpp
#include <vector>
#include "util.h"
#include <map>
#include <string>
#include <utility>
#include <memory>

using namespace std;

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
public:
    int removeElement(vector<int> &nums, int val) {
        auto fast = nums.begin();
        auto slow = nums.begin();
        while (fast != nums.end()) {
            if (*fast != val) {
                *slow = *fast;
                slow++;
            }
            fast++;
        }
        return static_cast<int>(distance(nums.begin(), slow));

    }
};
//leetcode submit region end(Prohibit modification and deletion)


int main(int argv, char *argc[]) {
}